apps/condo/domains/banking/schema/BankCategory.test.js

```diff
@@ -0,0 +1,166 @@
+
+                const client = await makeClientWithNewRegisteredAndLoggedInUser()
+
+                await expectToThrowAccessDeniedErrorToObj(async () => {
+                    await updateTestBankCategory(client, objCreated.id)
+                })
+            })
+
+            test('anonymous can\'t', async () => {
+                const admin = await makeLoggedInAdminClient()
+                const [objCreated] = await createTestBankCategory(admin)
+
+                const client = await makeClient()
+                await expectToThrowAuthenticationErrorToObj(async () => {
+                    await updateTestBankCategory(client, objCreated.id)
+                })
+            })
+        })
+
+        describe('hard delete', () => {
+            test('admin can\'t', async () => {
+                const admin = await makeLoggedInAdminClient()
+                const [objCreated] = await createTestBankCategory(admin)
+
+                await expectToThrowAccessDeniedErrorToObj(async () => {
+                    await BankCategory.delete(admin, objCreated.id)
+                })
+            })
+
+            test('user can\'t', async () => {
+                const admin = await makeLoggedInAdminClient()
+                const [objCreated] = await createTestBankCategory(admin)
+
+                const client = await makeClientWithNewRegisteredAndLoggedInUser()
+                await expectToThrowAccessDeniedErrorToObj(async () => {
+                    await BankCategory.delete(client, objCreated.id)
+                })
+            })
+
+            test('anonymous can\'t', async () => {
+                const admin = await makeLoggedInAdminClient()
+                const [objCreated] = await createTestBankCategory(admin)
+
+                const client = await makeClient()
+                await expectToThrowAccessDeniedErrorToObj(async () => {
+                    await BankCategory.delete(client, objCreated.id)
+                })
+            })
+        })
+
+        describe('read', () => {
+            test('admin can', async () => {
+                const admin = await makeLoggedInAdminClient()
+                const [obj, attrs] = await createTestBankCategory(admin)
+
+                const objs = await BankCategory.getAll(admin, {}, { sortBy: ['updatedAt_DESC'] })
+
+                expect(objs.length).toBeGreaterThanOrEqual(1)
+                expect(objs).toEqual(expect.arrayContaining([
+                    expect.objectContaining({
+                        id: obj.id,
+                        name: obj.name,
+                    }),
+                ]))
+            })
+
+            test('user can', async () => {
+                const admin = await makeLoggedInAdminClient()
+                const [obj] = await createTestBankCategory(admin)
+
+                const client = await makeClientWithNewRegisteredAndLoggedInUser()
+                const objs = await BankCategory.getAll(client, {}, { sortBy: ['updatedAt_DESC'] })
```

Comment:
It's better to pass id in where filter if you want your test result (`objs[0]`) to be predictable